src/modules/gauging-stations/controller.js

```diff
@@ -12,17 +15,56 @@ const getGaugingStation = async request => {
 };
 
 const getGaugingStationByRef = async request =>
-  controller.getEntities(
+  controllerHelper.getEntities(
     request.params.stationRef,
     gaugingStationService.getGaugingStationByRef
   );
 
 const getGaugingStationLicencesById = async request =>
-  controller.getEntities(
+  controllerHelper.getEntities(
     request.params.gaugingStationId,
     gaugingStationService.getGaugingStationLicencesById
   );
 
+const createLicenceGaugingStationLink = async request => {
+  try {
+    const { gaugingStationId } = request.params;
+    const {
+      licenceId,
+      licenceVersionPurposeConditionId,
+      thresholdUnit,
+      thresholdValue,
+      abstractionPeriod,
+      restrictionType,
+      alertType
+    } = request.payload;
+
+    // Check that the licence ID belongs to an actual licence.
+    await licencesService.getLicenceById(licenceId);
+
+    // Check that the gauging station ID belongs to an actual gauging station.
+    await gaugingStationService.getGaugingStation(gaugingStationId);
+
+    return licenceGaugingStationsService.createNewLicenceLink(gaugingStationId, licenceId, {
+      licenceVersionPurposeConditionId,
+      thresholdUnit,
+      thresholdValue,
+      ...abstractionPeriod,
```

Comment:
The abstraction period here is destructured whereas the `createNewLicenceLink()` default param indicates it is expecting an object so unclear if this will work.

The Joi validation rules on the route below look correct though.

Reply:
Flattened and corrected in 5c9be4f8